aws/vpc/main.tf

```diff
@@ -89,7 +89,27 @@ resource "aws_nat_gateway" "nat_gateway" {
 }
 
 /*
- * Routes for private subnets to use NAT gateway
+ * Set use_transit_gateway to true create transit gateway attachments in the private subnets and
+ * route traffic to the TGW instead of NAT GW.
+ * Should be used when create_nat_gateway=false.
+ */
+resource "aws_ec2_transit_gateway_vpc_attachment" "transit_gateway" {
+  count = var.use_transit_gateway ? 1 : 0
+  subnet_ids                                      = aws_subnet.private_subnet.*.id
+  transit_gateway_id                              = var.transit_gateway_id
+  vpc_id                                          = aws_vpc.vpc.vpc_id
+  transit_gateway_default_route_table_association = var.transit_gateway_default_route_table_association
+  transit_gateway_default_route_table_propagation = var.transit_gateway_default_route_table_propagation
+
+  tags = {
+    Name     = "TGW-${var.app_name}-${var.app_env}"
+    app_name = var.app_name
+    app_env  = var.app_env
+  }
+}
+
+/*
+ * Routes for private subnets to use NAT or Transit gateway
  */
 resource "aws_route_table" "nat_route_table" {
```

Comment:
@briskt Do you think this should be renamed, since it could be used with either a NAT or a TGW? There is another route table for the Internet Gateway, but I'm not sure what we would rename this _to_. Perhaps `private_route_table`? (We would probably need a bit of code here to handle mapping that rename, for existing systems using this terraform module, if we do a rename.)

Reply:
This may be a case where the [`moved`](https://developer.hashicorp.com/terraform/language/modules/develop/refactoring) directive could help avoid a breaking change. Without it, any module that uses this code would experience a destroy/create cycle on this resource.

```
moved {
  from = aws_route_table.nat_route_table
  to   = aws_route_table.private_route_table
}
```